### ISSUE TITLE
ASRock Z77 Extreme4 Will Sometimes Fail to Boot With Error 62

### ISSUE THREAD

**Issue by aolszowka** [xCuri0/ReBarUEFI#79]
**System**
- Motherboard: [ASRock Z77 Extreme4](https://www.asrock.com/mb/intel/z77%20extreme4/) - Modifications Start from Latest 3.00 Beta BIOS Available From ASRock
- GPU: [ASRock Intel Arc A380 Challenger ITX 6GB OC](https://www.asrock.com/Graphics-Card/Intel/Intel%20Arc%20A380%20Challenger%20ITX%206GB%20OC/)
- OS: Windows 10 22H2
 - [x] CSM is turned off
 - [x] 4G decoding is enabled
 - [x] UEFIPatch is applied (see [Using UEFIPatch](https://github.com/xCuri0/ReBarUEFI/wiki/Using-UEFIPatch) for more information). On some motherboards [DSDT Patching](https://github.com/xCuri0/ReBarUEFI/wiki/DSDT-Patching) is also needed
 - [x] I have read [Common issues (and fixes)](https://github.com/xCuri0/ReBarUEFI/wiki/Common-issues-(and-fixes))

**Description**
I was able to successfully modify the BIOS to allow ReBAR to work as per my post here: https://github.com/xCuri0/ReBarUEFI/issues/11#issuecomment-1605868162

However, every so often, most noticeably during a Windows Update "Install Updates and Restart" Cycle my board will fail to boot and display an Error 62 on the ASRock Motherboard Dr. Debug Readout. Looking online this error code relates to "Chipset initialization error." Clearing the CMOS and then restarting always has resolved the issue, but it just means I need to reapply my settings.

This doesn't appear to occur when a "normal" reboot happens or I shutdown and restart the machine. Again I have only noticed this on a Windows Update "Install Updates and Restart" Cycle or when automatic updates are executed. Any ideas on what might be different here?

I have the modified BIOS if that would help.

**Comment by xCuri0** [comment]
@aolszowka have you checked your CMOS battery voltage ? hwinfo should say it

**Comment by aolszowka** [comment]
@xCuri0 This is VBAT right? Seems to read at 3.248? I honestly don't know what expected range should be:

![image](https://github.com/xCuri0/ReBarUEFI/assets/2592881/0ab60fb5-480e-4bf0-a7bd-aa9d0e9f8f42)

I'd be happy to swap out the battery its just a CR2032; I don't think I've ever replaced this (build occurred in ~2011).

Is there something that leads you to think this might be the issue?

**Comment by xCuri0** [comment]
@aolszowka voltage looks ok

Maybe you could redo the BIOS modification but use MMTool to replace modules (pad file workaround) because I've heard UEFITool sometimes causes corruption like this.

**Comment by aolszowka** [comment]
@xCuri0 Thank you for the reply, honestly its not that big a deal and really only happens on Windows Updates. It seems so bizarre that an `Update and Restart` is somehow different than a normal restart. I'll keep this open until Patch Tuesday (2023/09/12) and see if it reproduces again if that is alright?

**Comment by aolszowka** [comment]
@xCuri0 Closing the loop on this, Patch Tuesday (2023/09/12) after installing Windows Updates and selecting `Update and Restart` I experienced the same issue. This machine rebooted (`Restart`) and Shutdown just fine. Absolutely bizarre. I am going to close out the issue as I have not encountered any other issues with this mod, but perhaps at some point in the future I'll be able to figure out just what is different between an `Update and Restart` and `Restart`. Next Patch Tuesday I am going to attempt an `Update and Shutdown` to see if that yields any different results.

**Comment by Naddly** [comment]
@aolszowka I hope I'm not bothering you, I have this same motherboard and the desire to have ReBar working but I have no experience with BIOS modification and I hope you can provide me with the already modified BIOS file if you still have it, accompanied by any tips that might be useful in the process. Thank you in advance. 😃

**Comment by aolszowka** [comment]
@Naddly I've got more details in my post here: https://github.com/xCuri0/ReBarUEFI/issues/11#issuecomment-1605868162

This part was critical I feel:

"During my entire process I actually had a CH341 Programmer available to me along with a bunch of spare W25Q64BVAIG's. The chip is socketed on this motherboard which makes swaps super simple and I always had a way to fall back when something went wrong. I had to use AsProgrammer_2.1.0.13_fix that I found off some YouTube guide, I picked up the programmer from Amazon and the spares from eBay from China (took 2 weeks to get here, but was worth it)."

Feel free to contact me @ my username from a mail service from Google.
